numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -219,7 +220,14 @@ typedef enum {
         /* Allow safe casts or casts within the same kind */
         NPY_SAME_KIND_CASTING=3,
         /* Allow any casts */
-        NPY_UNSAFE_CASTING=4
+        NPY_UNSAFE_CASTING=4,
+        /*
+         * Flag to allow signalling that a cast is a view, this flag is not
+         * valid when requesting a cast of specific safety.
+         * _NPY_CAST_IS_VIEW|NPY_EQUIV_CASTING means the same as NPY_NO_CASTING.
+         */
+        // TODO-DTYPES: Needs to be documented.
+        _NPY_CAST_IS_VIEW = 1 << 16,
```

Comment:
Thought I had commented here before...  Extending this struct is not ideal, but I like the path of returning the casting safety instead of asserting it (e.g. also allows adding new casting safety levels in theory). Further, indicating when a cast is a view seems helpful (no need to fetch a casting loop).
But, extending the enum is not ideal for two reasons:

1. The new fields are not useful for existing API which *requests* a certain casting safety rather than reporting it.
2. I don't want to make this public quite yet.

So, not quite sure how to best handle it? A new enum for now?

There are two other details:

1. `NPY_NO_CASTING` should be equivalent to be `NPY_EQUIV_CASTING | NPY_CAST_IS_VIEW).`, the code right now doesn't use that (I had that before, it probably is fine -- Void might have some corner cases, due to weird promotion rules, but right now everything seems to pan out.)
2. An error `-1` return cannot distinguish an impossible cast from a real error without `PyErr_Occurred()`, adding an `NPY_IMPOSSIBLE_CAST` may be better?

Reply:
The reason why I pointed to it is that this is the only public API change right now.  There is no definitely no issue with extending the enum itself.